stix_shifter_modules/cybereason/stix_transmission/results_connector.py

```diff
@@ -0,0 +1,94 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_results_connector import BaseResultsConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from json.decoder import JSONDecodeError
+import json
+
+
+class InvalidRequestException(Exception):
+    pass
+
+
+class InternalServerErrorException(Exception):
+    pass
+
+
+class ResultsConnector(BaseResultsConnector):
+    def __init__(self, api_client):
+
+        self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
+
+    def create_results_connection(self, query, offset, length):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :return: dict
+        """
+        return_obj = {}
+        response_dict = {}
+        try:
+            offset = int(offset)
+            length = int(length)
+            if isinstance(query, dict):
+                query = json.dumps(query)
+            response_wrapper = self.api_client.get_search_results(query)
+            if response_wrapper.code == 200:
+                return_obj['success'] = True
+            elif 399 < response_wrapper.code < 500:
+                raise InvalidRequestException(response_wrapper.response.text)
+            elif response_wrapper.code == 500:
+                raise InternalServerErrorException(response_wrapper.response.text)
+            response_dict = json.loads(response_wrapper.read().decode('utf-8'))
+            results = self.get_results_data(response_dict)
+            return_obj['data'] = results[offset:length]
+
+            # session log out
+            response_wrapper = self.api_client.session_log_out(response_wrapper)
+            if not response_wrapper.code == 200:
+                raise InvalidRequestException(response_wrapper.response.text)
+
+        except JSONDecodeError:
+            response_dict['type'] = "AuthenticationError"
+            response_dict['message'] = "Authentication Invalid"
```

Comment:
this `JSONDecodeError` exception handling is misleading. that exception supposed to be handling response data format error not authentication. I understand why you are mapping this exception to authentication error but there should be a different indicator if not I would suggest to put some comments in the code.

Reply:
Code changes has been done to throw InvalidAuthenticationException by checking history response code '302' ( Invalid authentication) in connector.py file.